src/reports/library/sonatareport.cpp

```diff
@@ -29,7 +29,7 @@ bool SonataReport::is_empty() {
     return m_reports.empty();
 }
 
-std::shared_ptr<Report> SonataReport::create_report(
+const std::shared_ptr<Report>& SonataReport::create_report(
```

Comment:
why are you returning a const-ref to the `shared_ptr` ? This doesn't increase the ref-counter in the `shared_ptr` so couldn't the Report be destroyed while still being pointed at? Probably I'm again messing up something. You have more instances of this return-by-ref in this PR...

Reply:
You are right.
It is explained here: https://stackoverflow.com/questions/10643563/how-to-return-smart-pointers-shared-ptr-by-reference-or-by-value
Apparently the penalty of passing by value is not so big with new compilers.